packages/client/components/TeamPrompt/WorkDrawer/GCalEventCard.tsx

```diff
@@ -0,0 +1,120 @@
+import clsx from 'clsx'
+import {ContentCopy} from '@mui/icons-material'
+
+interface Props {
+  eventRef: GCalEventCard_event$key
+}
+
+const formatTime = (time: Date, excludeAmPm?: boolean | null) => {
+  return (
+    time
+      .toLocaleString('en-US', {hour: 'numeric', minute: 'numeric', hour12: true})
+      // 'XX:YY AM' -> 'XX:YYam'
+      .replace(/ (PM|AM)/, excludeAmPm ? '' : '$1')
+      .toLowerCase()
+  )
+}
+
+const getDayDifference = (startDate: Date, endDate: Date) => {
+  const startDateCopy = new Date(startDate)
+  const endDateCopy = new Date(endDate)
+  const startDay = new Date(startDateCopy.setHours(0, 0, 0, 0))
+  const endDay = new Date(endDateCopy.setHours(0, 0, 0, 0))
+  return (endDay.getTime() - startDay.getTime()) / (1000 * 60 * 60 * 24)
+}
+
+const GCalEventCard = (props: Props) => {
+  const {eventRef} = props
+
+  const result = useFragment(
+    graphql`
+      fragment GCalEventCard_event on GcalIntegrationEvent {
+        summary
+        startDate
+        endDate
+        link
+      }
+    `,
+    eventRef
+  )
+
+  const {tooltipPortal, openTooltip, closeTooltip, originRef} = useTooltip<HTMLDivElement>(
+    MenuPosition.UPPER_CENTER
+  )
+
+  const {
+    tooltipPortal: copiedTooltipPortal,
+    openTooltip: openCopiedTooltip,
+    closeTooltip: closeCopiedTooltip,
+    originRef: copiedTooltipRef
+  } = useTooltip<HTMLDivElement>(MenuPosition.LOWER_CENTER)
+
+  const handleCopy = () => {
+    openCopiedTooltip()
+    setTimeout(() => {
+      closeCopiedTooltip()
+    }, 2000)
+  }
+
+  const startDate = result.startDate ? new Date(result.startDate) : null
+  const endDate = result.endDate ? new Date(result.endDate) : null
+
+  const dayDifference = startDate && endDate ? getDayDifference(startDate, endDate) : 0
+
+  // Always show am/pm for start date unless:
+  // 1. start + end times are within 3 hours of each other.
+  // 2. AND start + end times are either both in the AM or both in the PM.
+  const shouldAlwaysShowAmPm =
+    endDate &&
+    startDate &&
+    endDate.getTime() - startDate.getTime() < 3 * 60 * 60 * 1000 &&
+    ((endDate.getHours() >= 12 && startDate.getHours() >= 12) ||
+      (endDate.getHours() < 12 && startDate.getHours() < 12))
+
+  return (
+    <div className='group'>
+      <div
+        className={clsx('rounded border border-solid border-slate-300 p-4 hover:border-slate-600')}
+      >
+        <div>
+          <a
+            className='hover:underline'
+            href={result.link ?? undefined}
+            target='_blank'
+            rel='noreferrer'
+          >
+            {result.summary}
+          </a>
+        </div>
+        <div className='flex justify-between text-sm text-slate-600'>
+          {startDate && `${formatTime(startDate, shouldAlwaysShowAmPm)}`}
+          {startDate && endDate && ' - '}
+          {endDate && `${formatTime(endDate)}`}
+          {dayDifference > 0 && ` (+${dayDifference}d)`}
```

Comment:
+1 this is a little bit hard to read, what about something like

```
const formatEventTime = (startDate: Date | null, endDate: Date | null, dayDifference: number, shouldAlwaysShowAmPm: boolean) => {
  let formattedTime = '';
  
  if (startDate) {
    formattedTime += formatTime(startDate, shouldAlwaysShowAmPm);
  }

  if (startDate && endDate) {
    formattedTime += ' - ';
  }

  if (endDate) {
    formattedTime += formatTime(endDate);
  }

  if (dayDifference > 0) {
    formattedTime += ` (+${dayDifference}d)`;
  }

  return formattedTime;
}
```

Reply:
Nice call, definitely an improvement